libs/libarchfpga/src/read_xml_arch_file.cpp

```diff
@@ -4156,10 +4156,10 @@ static void ProcessDirects(pugi::xml_node Parent, t_direct_inf** Directs, int* N
         }
 
         /* Check that the direct chain connection is not zero in both direction */
```

Comment:
Delete this code and the comment.  If you think it's appropriate, you can note that directs can be back to themselves in some comment.